.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,79 @@
+{
+	"name": "Home Ops",
+	"image": "mcr.microsoft.com/devcontainers/base:debian",
+	"containerEnv": {
+		"KUBECONFIG": "${containerWorkspaceFolder}/kubeconfig",
+		"SOPS_AGE_KEY_FILE": "${containerWorkspaceFolder}/.age.key"
+	},
+	"postCreateCommand": {
+		"deps": "task deps"
+	},
+	"features": {
+		// Upgrade OS packages
+		"ghcr.io/devcontainers/features/common-utils:2": {
+			"version": "latest",
+			"installZsh": false,
+			"installOhMyZsh": false
+		},
+		"ghcr.io/devcontainers-contrib/features/go-task:1": {
+			"version": "3.31.0"
+		},
+    "ghcr.io/devcontainers-contrib/features/ansible:2": {
+			"version": "2.16.0"
+		},
+		"ghcr.io/devcontainers-contrib/features/age:1": {
+			"version": "1.1.0-rc.1"
+		},
+    "ghcr.io/audacioustux/devcontainers/cilium:1": {},
+    "ghcr.io/devcontainers-contrib/features/cloudflared:1": {
+			"version": "2021.10.0"
+		},
+    "ghcr.io/jsburckhardt/devcontainer-features/flux:1": {
+			"version": "v2.1.2"
+		},
+    "ghcr.io/devcontainers/features/kubectl-helm-minikube:1": {
+      "version": "1.28.3",
+      "helm": "latest",
+      "minikube": "none"
+    },
+    "ghcr.io/rio/features/kustomize:1": {
+      "version": "latest"
+    },
+		"ghcr.io/devcontainers-contrib/features/sops:1": {
+			"version": "3.7.3"
+		},
+    "ghcr.io/dhoeric/features/stern:1": {
+      "version": "1.27.0"
+    },
+    "ghcr.io/eitsupi/devcontainer-features/jq-likes:2": {
+      "jqVersion": "1.7"
+    }
+	},
+	"customizations": {
+		"vscode": {
+			"settings": {
+				"terminal.integrated.profiles.linux": {
+					"bash": {
+						"path": "/bin/bash"
+					}
+				},
+				"terminal.integrated.defaultProfile.linux": "bash"
+			},
+			"extensions": [
+				"albert.TabOut",
+				"britesnow.vscode-toggle-quotes",
+				"EditorConfig.editorconfig",
+				"fcrespo82.markdown-table-formatter",
+				"GitHub.copilot",
+				"mikestead.dotenv",
+				"mitchdenny.ecdc",
+				"PKief.material-icon-theme",
+				"redhat.ansible",
+				"signageos.signageos-vscode-sops",
+				"task.vscode-task",
+				"VisualStudioExptTeam.vscodeintellicode",
+				"will-stone.in-any-case"
+			]
```

Comment:
Can these be pulled from the .vscode folder? It would be nice to not have to defined them twice

Reply:
I think the only option is a simple array per the docs https://containers.dev/supporting#visual-studio-code. Could make an actions workflow to keep the lists in sync but that is also probably overkill.

Reply:
Probably should only preinstall a subset of very useful extensions.

Reply:
After considering the python feature surprise extensions. I'm now leaning towards no preinstalled extensions but maybe that is not the goal for the template project.